### PR TITLE
style: use new syntax for typing

### PR DESCRIPTION
**Pull request by prusnak**
Closes https://github.com/trezor/trezor-firmware/issues/1314

Let's not merge before https://github.com/trezor/trezor-firmware/pull/1292 is in, so there are no unnecessary conflicts for that PR

### CODE REVIEW

**Comment by prusnak** [comment]
We still need to modify the `pb2py` script and regenerate the messages, but there is one issue with that that we need to discuss first.

File: core/src/apps/ethereum/keychain.py [hunk 1]
```diff
@@ -16,7 +16,7 @@
     from apps.common.keychain import MsgOut, Handler, HandlerWithKeychain
 
     class MsgWithAddressN(MessageType, Protocol):
-        address_n = ...  # type: List[int]
+        address_n: List[int] = ...
```
Comment (matejcik) [inline core/src/apps/ethereum/keychain.py:19]: the `= ...` is superfluous when the type annotation is present

File: python/src/trezorlib/toif.py [hunk 2]
```diff
@@ -63,9 +63,9 @@ def _to_grayscale(data: bytes) -> bytes:
 
 @attr.s
 class Toif:
-    mode = attr.ib()  # type: firmware.ToifMode
-    size = attr.ib()  # type: Tuple[int, int]
-    data = attr.ib()  # type: bytes
+    mode: firmware.ToifMode = attr.ib()
```
Comment (matejcik) [inline python/src/trezorlib/toif.py:66]: we could get rid of `= attr.ib()` by enabling `@attr.s(auto_attribs=True)`

File: core/mocks/uctypes.pyi [hunk 3]
```diff
@@ -1,25 +1,25 @@
...
-BFUINT8 = ...  # type: int
-BFINT8 = ...  # type: int
-BFUINT16 = ...  # type: int
-BFINT16 = ...  # type: int
-BFUINT32 = ...  # type: int
-BFINT32 = ...  # type: int
-BF_POS = ...  # type: int
-BF_LEN = ...  # type: int
-FLOAT32 = ...  # type: int
+ARRAY: int = ...
```
Comment (matejcik) [inline core/mocks/uctypes.pyi:1]: `= ...` is superfluous

File: core/src/apps/bitcoin/keychain.py [hunk 4]
```diff
@@ -15,7 +15,7 @@
     from .authorization import CoinJoinAuthorization
 
     class MsgWithCoinName(Protocol):
-        coin_name = ...  # type: str
+        coin_name: str = ...
```
Comment (matejcik) [inline core/src/apps/bitcoin/keychain.py:18]: `= ...` is superfluous

**Review by matejcik — CHANGES_REQUESTED** [review]

File: core/src/apps/bitcoin/keychain.py (hunk 4 above)
Comment (prusnak) [reply core/src/apps/bitcoin/keychain.py]: Addressed in 998eddfeba30f43c6c550f01b717c0c654650ac2

File: core/src/apps/ethereum/keychain.py (hunk 1 above)
Comment (prusnak) [reply core/src/apps/ethereum/keychain.py]: Addressed in 998eddfeba30f43c6c550f01b717c0c654650ac2

File: core/mocks/uctypes.pyi (hunk 3 above)
Comment (prusnak) [reply core/mocks/uctypes.pyi]: Addressed in 7a39775f0adae58808ca22a8cab29d8ef5004717

File: python/src/trezorlib/toif.py (hunk 2 above)
Comment (prusnak) [reply python/src/trezorlib/toif.py:66]: Let's do this in some other PR, please

**Comment by prusnak** [comment]
I also rewrote the autogenerated protobuf enums in 0cd4ed5ace35eb524896fb99927adb6dd0ceba12 - can you please @matejcik review this as well?

**Review by matejcik — APPROVED** [review]
lgtm

we could experiment with using `Final` for the enum types, but that's for another PR, perhaps #1237 will do something about this too

**Comment by prusnak** [comment]
@matejcik any idea how to fix the failing tests? e.g. https://gitlab.com/satoshilabs/trezor/trezor-firmware/-/jobs/798671402

**Comment by matejcik** [comment]
I guess we'll exclude the name `"Literal"` explicitly?
https://github.com/trezor/trezor-firmware/blob/master/python/src/trezorlib/mapping.py#L27-L29

**Pull request merged**